HPXMLtoOpenStudio/tests/test_defaults.rb

```diff
@@ -1163,117 +1163,153 @@ def test_central_air_conditioners
     hpxml_bldg.cooling_systems[0].cooling_efficiency_seer2 = 11.4
     XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
     _default_hpxml, default_hpxml_bldg = _test_measure()
-    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, -0.11, -0.22, 12345, 12.0, 40.0, 1.0)
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, 780, -0.11, -0.22, 12345, 12.0, 40.0, 1.0)
 
     # Test autosizing with factors
     hpxml_bldg.cooling_systems[0].cooling_capacity = nil
     hpxml_bldg.cooling_systems[0].cooling_autosizing_factor = 1.2
     XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
     _default_hpxml, default_hpxml_bldg = _test_measure()
-    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, -0.11, -0.22, nil, 12.0, 40.0, 1.2)
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, 780, -0.11, -0.22, nil, 12.0, 40.0, 1.2)
+
+    # Test defaults - airflow
+    hpxml_bldg.cooling_systems[0].cooling_capacity = 12345
+    hpxml_bldg.cooling_systems[0].cooling_autosizing_factor = nil
+    hpxml_bldg.cooling_systems[0].cooling_airflow_cfm = nil
+    hpxml_bldg.cooling_systems[0].airflow_defect_ratio = nil
+    XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+    _default_hpxml, default_hpxml_bldg = _test_measure()
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, 513, -0.11, 0, 12345, 12.0, 40.0, 1.0)
+
+    # Test defaults - ductwork restriction low
+    hpxml_bldg.cooling_systems[0].max_cooling_airflow_cfm = 1000
+    XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+    _default_hpxml, default_hpxml_bldg = _test_measure()
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.66, 513, -0.11, 0, 12345, 12.0, 40.0, 1.0)
+
+    # Test defaults - ductwork restriction high
+    hpxml_bldg.cooling_systems[0].max_cooling_airflow_cfm = 100
+    XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+    _default_hpxml, default_hpxml_bldg = _test_measure()
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.025, 100, -0.11, 0, 2405, 12.0, 40.0, 1.0)
+
+    # Test defaults - ductwork restriction autosized
+    hpxml_bldg.cooling_systems[0].cooling_capacity = nil
+    XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+    _default_hpxml, default_hpxml_bldg = _test_measure()
+    _test_default_central_air_conditioner_values(default_hpxml_bldg.cooling_systems[0], 0.88, HPXML::HVACCompressorTypeVariableSpeed, 0.008, 100, -0.11, 0, 2405, 12.0, 40.0, 1.0)
```

Comment:
For central AC, furnace, and ASHP, we check:
* the autosized airflow with no restriction and fixed capacity
* the autosized airflow with a high restriction and fixed capacity
* the autosized airflow with a low restriction and fixed capacity
* the autosized airflow with a low restriction and autosized capacity